### PR TITLE
Minor fixes in AbstractStorageProviderNamespaceAdmin for namespaces root directory

### PR DESCRIPTION
**Pull request by rohitsinha54**
Issue: https://issues.cask.co/browse/CDAP-6743
https://issues.cask.co/browse/CDAP-6716

Build: http://builds.cask.co/browse/CDAP-DUT4533-5

### CODE REVIEW

**Comment by gokulavasan** [comment]
LGTM 👍

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java [hunk 1]
```diff
@@ -106,9 +107,7 @@ private void deleteLocation(NamespaceId namespaceId) throws IOException {
         LOG.debug("Custom location mapping %s was found while deleting namespace %s. Deleting all data inside it but" +
                     "skipping namespace home directory delete.", namespaceHome, namespaceId);
         // delete everything inside the namespace home but not the namespace home as its user owned directory
-        for (Location childLocation : namespaceHome.list()) {
-          childLocation.delete(true);
-        }
+        Locations.deleteContent(namespaceHome);
```
Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java:110]: Just checking, do we impersonate while doing this?

**Comment by rohitsinha54** [comment]
Waiting on build.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java (hunk 1 above)
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java:110]: yes; DefaultNamespaceStorageHandler calls this method with impersonation, I believe.

**Pull request merged**